tools/walletextension/useraccountmanager/user_account_manager.go

```diff
@@ -47,6 +51,8 @@ func (m *UserAccountManager) AddAndReturnAccountManager(userID string) *accountm
 // (we are not loading all of them at startup to limit the number of established connections)
 // If a UserAccountManager does not exist for the userID, it returns nil and an error.
 func (m *UserAccountManager) GetUserAccountManager(userID string) (*accountmanager.AccountManager, error) {
+	m.mu.Lock()
```

Comment:
Worth using RWMutex and just doing read lock for this read op?

Reply:
Good point, using `RWMutex` and `m.mu.RLock()` here.